gcp/src/test/java/org/apache/iceberg/gcp/biglake/BigLakeCatalogTest.java

```diff
@@ -0,0 +1,410 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+public class BigLakeCatalogTest extends CatalogTests<BigLakeCatalog> {
+
+  @Rule public final MockitoRule mockito = MockitoJUnit.rule();
+  @TempDir public Path temp;
+
+  private static final String GCP_PROJECT = "my-project";
+  private static final String GCP_REGION = "us";
+  private static final String CATALOG_ID = "biglake";
+
+  private String warehouseLocation;
+
+  // For tests using a BigLake catalog connecting to a mocked service.
+  private MockMetastoreService mockMetastoreService;
+  private MockServiceHelper mockServiceHelper;
+  private LocalChannelProvider channelProvider;
+  private BigLakeCatalog bigLakeCatalogUsingMockService;
+
+  // For tests using a BigLake catalog with a mocked client.
+  private BigLakeClient mockBigLakeClient;
+  private BigLakeCatalog bigLakeCatalogUsingMockClient;
+
+  @BeforeEach
+  public void before() throws Exception {
+    mockMetastoreService = new MockMetastoreService();
+    mockServiceHelper =
+        new MockServiceHelper(
+            UUID.randomUUID().toString(), Arrays.<MockGrpcService>asList(mockMetastoreService));
+    mockServiceHelper.start();
+
+    File warehouse = temp.toFile();
+    warehouseLocation = warehouse.getAbsolutePath();
+
+    ImmutableMap<String, String> properties =
+        ImmutableMap.of(
+            BigLakeCatalog.PROPERTIES_KEY_GCP_PROJECT,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation);
+
+    channelProvider = mockServiceHelper.createChannelProvider();
+    MetastoreServiceSettings settings =
+        MetastoreServiceSettings.newBuilder()
+            .setTransportChannelProvider(channelProvider)
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .build();
+
+    bigLakeCatalogUsingMockService = new BigLakeCatalog();
+    bigLakeCatalogUsingMockService.setConf(new Configuration());
+    bigLakeCatalogUsingMockService.initialize(
+        CATALOG_ID,
+        properties,
+        GCP_PROJECT,
+        GCP_REGION,
+        new BigLakeClient(settings, GCP_PROJECT, GCP_REGION));
+
+    mockBigLakeClient = mock(BigLakeClient.class);
+    bigLakeCatalogUsingMockClient = new BigLakeCatalog();
+    bigLakeCatalogUsingMockClient.setConf(new Configuration());
+    bigLakeCatalogUsingMockClient.initialize(
+        CATALOG_ID, properties, GCP_PROJECT, GCP_REGION, mockBigLakeClient);
+  }
+
+  @AfterEach
+  public void after() {
+    mockServiceHelper.stop();
+  }
+
+  @Override
+  protected boolean requiresNamespaceCreate() {
+    return true;
+  }
+
+  @Override
+  protected BigLakeCatalog catalog() {
+    return bigLakeCatalogUsingMockService;
+  }
+
+  @Override
+  protected boolean supportsNamesWithSlashes() {
+    return false;
+  }
+
+  @Test
+  public void testNamespaceWithSlash() {
```

Comment:
This should be unnecessary because of the `supportsNamesWithSlashes` above. If you want to test BigLake specific behavior, create a new test case instead.

Reply:
I removed the slash tests.